### DISCUSSION TITLE
Copilot inline suggestions not closing brackets

### DISCUSSION THREAD

**Discussion by WilsontheWolf** [community/community#7656] [Copilot Conversations]
When using copilot in vscode, the inline suggestions often don't have closing brackets. This becomes a pain, because then I need to manually close them when writing. If we could have it where they respected our ide's auto bracket. This becomes more of a pain when I am doing something such as writing functions inside of other function calls where as whats closed and not closed becomes more of a mess.

Here is a simple example of this.

https://user-images.githubusercontent.com/33164598/141597916-229e708b-d77b-47ac-adfe-16a394ae2895.mp4

 In this example, the { of the function should be auto closed when I insert the code. If this could be remidied that would be lovely. Thanks!

**Comment by MulleDK19** [comment]
No, that would be terrible. It already suggests the end bracket when you get to that line.

If it inserted a } automatically to close the function, which is not what it was completing, you'd have to constantly remove closing brackets on every suggestion when you don't need the function to end. It has no way of knowing whether there's any more to come in that function or if you're done. It only completes what it's asked to, which is how it should be. It shouldn't start closing functions or the outer type, etc. that would be an absolute nightmare...

If you want it to complete add the }, then let it complete the entire function, don't edit it. Change the comment to say "using fetch" or something.

**Reply by vincentor** [reply]
I don't agree with that. Function's bracket will be close eventually, function show as a closing block, if copilot close the bracket and move the cursor into the bracket block, it will be nice and won't cause any trouble at all.

**Reply by ToonLunk** [reply]
This is now what it does, and I indeed am constantly deleting curly braces...

**Comment by GeneGenie** [comment]
Double voting on this. Why does it close freaking curly brace without closing round ?

**Comment by hansSchall** [comment]
The current behavior is really anoying, sometimes it takes me longer to find and fix the missing brackets than it would take to write the code without copilot but with auto-closing brackets.

**Comment by rhys-vdw** [comment]
I'm having the opposite problem all of a sudden, it's closing blocks which I do not want.
